logging.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+import Control.Applicative ((<|>))
+import Control.Exception.Safe (catchAny, displayException)
+import Control.Monad (guard)
+import Data.Foldable (fold)
+import System.Directory (doesFileExist)
+import System.IO (hPutStr, stderr)
+
+
+data Log =
+  Log {
+    logInfo :: String -> IO (),
+    logError :: String -> IO ()
+  }
+
+
+-- A log that writes to the console.
+consoleLog =
+  Log {
+    logInfo = putStr,
+    logError = hPutStr stderr
+  }
+
+
+-- A log that writes to the specified files.
+fileLog infoPath errPath =
+  Log {
+    logInfo = appendFile infoPath,
+    logError = appendFile errPath
```

Comment:
What's the default BufferMode for appendFile? Should I force this to flush, or change the buffer mode?